lua/cfc_random_spawn/module/sv_player_spawning.lua

```diff
@@ -217,20 +297,28 @@ local function getPopularCenter( plys )
     return bestCenter
 end
 
+local function updatePopularCenter( measurablePlayers )
+    if CFCRandomSpawn.doDynamicCenters then
+        mostPopularCenter = getDynamicPvpCenter( measurablePlayers )
+    else
+        mostPopularCenter = getPopularCenter( measurablePlayers )
+    end
+    CFCRandomSpawn.mostPopularCenter = mostPopularCenter
+
+    CENTER_CUTOFF = mostPopularCenter.overrideCutoff or DEFAULT_CENTER_CUTOFF
+    CENTER_CUTOFF_SQR = mostPopularCenter.overrideCutoffSqr or DEFAULT_CENTER_CUTOFF_SQR
+end
+
 function CFCRandomSpawn.getOptimalSpawnPos()
     local measurablePlayers = getMeasurablePlayers()
     local allLivingPlys = getLivingPlayers()
 
-    if CENTER_UPDATE_ON_RESPAWN then
-        mostPopularCenter = getPopularCenter( measurablePlayers ) or pvpCenters[1]
-        CFCRandomSpawn.mostPopularCenter = mostPopularCenter
-
-        CENTER_CUTOFF = mostPopularCenter.overrideCutoff or DEFAULT_CENTER_CUTOFF
-        CENTER_CUTOFF_SQR = mostPopularCenter.overrideCutoffSqr or DEFAULT_CENTER_CUTOFF_SQR
+    if CENTER_UPDATE_ON_RESPAWN or not CFCRandomSpawn.mostPopularCenter then
+        updatePopularCenter( measurablePlayers )
```

Comment:
could we just get rid of the option to update center on respawn? is there every a time we will want to do that?

Reply:
👍